### PR TITLE
Add courses

### PR DESCRIPTION
**Pull request by cutieprobe**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -18,5 +17,9 @@
     "camelcase": "off",
     "no-param-reassign": "warn",
     "arrow-parens": ["error", "as-needed"]
-  }
+  },
+  "extends": [
+    "airbnb-base",
+    "prettier"
```
Comment (gary-lgy) [inline .eslintrc.json:23]: Why?

File: views/partials/header.ejs [hunk 2]
```diff
@@ -42,6 +42,10 @@
       <%=user.name%>
     </button>
     <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
+      <a class="dropdown-item" href="/courses/new">
```
Comment (gary-lgy) [inline views/partials/header.ejs:45]: This should probably be available in the courses index page as well.

**Review by Na-Nazhou — CHANGES_REQUESTED** [review]
image

Course Edit and Delete seem not working. Might be related to the bug in #41.

**Comment by Na-Nazhou** [comment]
A possible enhancement to the course INDEX page is to group courses by semester (one course table for each semester) and order the semesters by start time. 

This can be done in another pr:)

File: .eslintrc.json (hunk 1 above)
Comment (cutieprobe) [reply .eslintrc.json:23]: There is conflict between eslint and prettier. e.g. add line break before/after operators

Comment (gary-lgy) [reply .eslintrc.json:23]: Can't reproduce. Is there a concrete example?

Comment (gary-lgy) [reply .eslintrc.json:23]: Nevermind. I think it's good to have it either way.

Comment (cutieprobe) [reply .eslintrc.json:23]: For the highlighted line, it exceed maximum line length and has to break into 2 lines. Eslint require the plus sign to be at the front of line 10 whereas Prettier requires it at the back of line 9.
operator_linebreak

File: views/courseEdit.ejs [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<%- contentFor('content') %>
+<div class="container col-md-4 col-md-offset-4">
+  <h1 class="text-center">Update Semester</h1>
+  <form id="course_form" action="/courses/<%=course.semester_name%>/<%=course.module_code%>" role="form" method="post">
```
Comment (Na-Nazhou) [inline views/courseEdit.ejs]:
```suggestion
  <form id="course_form" action="/courses/<%=encodeURIComponent(course.semester_name)%>/<%=encodeURIComponent(course.module_code)%>" role="form" method="post">
```

Use encodeURIComponent(), then there is no need to use * in the routes. Refer to #42

File: views/courseEdit.ejs [hunk 4]
```diff
@@ -0,0 +1,70 @@
+<%- contentFor('content') %>-
+<div class="container col-md-4 col-md-offset-4">
+  <h1 class="text-center">Update Semester</h1>
+  <form id="course_form"
+    action="/courses/<%=encodeURIComponent(course.semester_name)%>/<%=encodeURIComponent(course.module_code)%>"
+    role="form" method="post">
+    <div class="form-group">
+      <label for="name">Semester Name</label>
```
Comment (Na-Nazhou) [inline views/courseEdit.ejs:8]:
```suggestion
      <label for="semester_name">Semester Name</label>
```
Same for label fields below

**Review by Na-Nazhou — APPROVED** [review]

**Pull request merged**